Sources/SwiftCompilerPluginMessageHandling/CompilerPluginMessageHandler.swift

```diff
@@ -105,10 +126,18 @@ public class CompilerPluginMessageListener<Connection: MessageConnection, Provid
   }
 }
 
-/// 'CompilerPluginMessageHandler' is a type that handle a message and do the
-/// corresponding operation.
+/// A type that handles a plugin message and returns a response.
+///
+/// - SeeAlso: ``PluginProviderMessageHandler``
+@_spi(PluginMessage)
+public protocol PluginMessageHandler {
+  /// Handles a single message received from the plugin host.
+  func handleMessage(_ message: HostToPluginMessage) -> PluginToHostMessage
+}
```

Comment:
I guess this abstraction is needed anyway, so this is okay. But  TBH, I'm not a fan of `WasmInterceptingMessageHandler` in https://github.com/apple/swift/pull/73031.

Reply:
Yeah I'm not the biggest fan either :/ but that's the only approach I could think of which allows us to send _some_ messages from `swift-plugin-server` to `LibraryPluginProvider` while intercepting others. The alternative is to have an entirely separate `swift-wasm-plugin-server` (so that the primary plugin server can use `LibraryPluginProvider` directly) but both Doug and Yuta feel that having a unified plugin server is preferable (see https://github.com/apple/swift/pull/73031#discussion_r1580502591), and I agree that the benefits outweigh the downsides atm.

Curious to hear if you have any alternative suggestions though.